torchgeo/datasets/deepglobelandcover.py

```diff
@@ -0,0 +1,264 @@
+                mask_path = os.path.join(root, self.data_root, split_folder, "masks", str(id)+"_mask.png")
+                if os.path.exists(mask_path):
+                    self.files.append(dict(image=image_path, mask=mask_path))
+
+    def __getitem__(self, index: int) -> Dict[str, Tensor]:
+        """Return an index within the dataset.
+
+        Args:
+            index: index to return
+
+        Returns:
+            data and label at that index
+        """
+        image = self._load_image(index)
+        mask = self._load_target(index)
+        sample = {"image": image, "mask": mask}
+
+        if self.transforms is not None:
+            sample = self.transforms(sample)
+
+        return sample
+
+    def __len__(self) -> int:
+        """Return the number of data points in the dataset.
+
+        Returns:
+            length of the dataset
+        """
+        return len(self.files)
+
+    def _load_image(self, index: int) -> Tensor:
+        """Load a single image.
+
+        Args:
+            index: index to return
+
+        Returns:
+            the image
+        """
+        path = self.files[index]["image"]
+
+        with Image.open(path) as img:
+            array: "np.typing.NDArray[np.int_]" = np.array(img)
+            tensor = torch.from_numpy(array)
+            # Convert from HxWxC to CxHxW
+            tensor = tensor.permute((2, 0, 1))
+            return tensor
+
+
+
+    def _load_target(self, index: int) -> Tensor:
+        """Load the target mask for a single image.
+
+        Args:
+            index: index to return
+
+        Returns:
+            the target mask
+        """
+        path = self.files[index]["mask"]
+        with Image.open(path) as img:
+            array: "np.typing.NDArray[np.uint8]" = np.array(img) 
+            array = rgb_to_mask(array, self.colormap)
+            tensor = torch.from_numpy(array)
+            # Convert from HxWxC to CxHxW
+            tensor = tensor.to(torch.long)
+        return tensor
+
+    def _verify(self) -> None:
+        """Verify the integrity of the dataset.
+
+        Raises:
+            RuntimeError: if checksum fails or the dataset is not downloaded
+        """
+        # Check if the files already exist
+        if os.path.exists(os.path.join(self.root, self.data_root)):
+            return
+
+        # Check if .zip file already exists (if so extract)
+        filepath = os.path.join(self.root, self.filename)
+    
+        if os.path.isfile(filepath):
+            if self.checksum and not check_integrity(filepath, self.md5):
+                raise RuntimeError("Dataset found, but corrupted.")
+            extract_archive(filepath)
+            return
+
+        # Check if the user requested to download the dataset
+        raise RuntimeError(
+            "Dataset not found in `root` directory, either specify a different"
+            + " `root` directory or manually download the dataset to this directory."
+        )
+
+    def plot(
```

Comment:
Can you add a saved figure to the PR description so we can make sure the plot looks correct?

Reply:
I've shared it with Caleb! I can add it here too.